code/modules/mod/mod_control.dm

```diff
@@ -278,7 +278,7 @@
 			balloon_alert(wearer, "retract parts first!")
 			playsound(src, 'sound/machines/scanbuzz.ogg', 25, FALSE, SILENCED_SOUND_EXTRARANGE)
 			return
-	if(!wearer.incapacitated())
+	if(!wearer.incapacitated() && !active) // SKYRAT EDIT - MODSUIT ASSERT FAILURE FIX
```

Comment:
I think it's a better idea to make it obvious that you can't remove the MODsuit when it's active (and you should include that in your changelog entry too), rather than leaving it up to the user to figure out why they can't remove it from their back anymore. I know it's a few more lines, and you *will* have to test this again to make sure that it works (because I didn't), but I think that ultimately it should be better.
```suggestion

	// SKYRAT EDIT ADDITION START - Can't remove your MODsuit from your back when it's still active (as it can cause runtimes and even the MODsuit control unit to delete itself)
	if(!active)
		if(!wearer.incapacitated())
			balloon_alert(wearer, "deactivate first!")
			playsound(src, 'sound/machines/scanbuzz.ogg', 25, FALSE, SILENCED_SOUND_EXTRARANGE)

		return
	// SKYRAT ADDITION END
		
	if(!wearer.incapacitated())
```